extensions/notebook/src/test/book/book.test.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as should from 'should';
+import * as TypeMoq from 'typemoq';
+import * as path from 'path';
+import * as fs from 'fs';
+import * as rimraf from 'rimraf';
+import { BookTreeViewProvider } from '../../book/bookTreeView';
+import { BookTreeItem } from '../../book/bookTreeItem';
+
+describe('BookTreeViewProvider.getChildren', function (): void {
+	const rootFolderPath = path.join(__dirname, 'testBook');
+	const dataFolderPath = path.join(rootFolderPath, '_data');
+	const contentFolderPath = path.join(rootFolderPath, 'content');
+	const configFile = path.join(rootFolderPath, '_config.yml');
+	const tableOfContentsFile = path.join(dataFolderPath, 'toc.yml');
+	const notebookFile = path.join(contentFolderPath, 'notebook.ipynb');
+	const markdownFile = path.join(contentFolderPath, 'markdown.md');
+	const expectedNotebook = {
+		title: 'Notebook',
+		url: '/notebook'
+	};
+	const expectedMarkdown = {
+		title: 'Markdown',
+		url: '/markdown'
+	};
+	const expectedExternalLink = {
+		title: 'GitHub',
+		url: 'https://github.com/',
+		external: true
+	};
+	const expectedBook = {
+		sections: [expectedNotebook, expectedMarkdown, expectedExternalLink],
+		title: 'Test Book'
+	};
+
+	let mockExtensionContext: TypeMoq.IMock<vscode.ExtensionContext>;
+	let bookTreeViewProvider: BookTreeViewProvider;
+	let book: BookTreeItem;
+
+	this.beforeAll(async () => {
+		fs.mkdirSync(rootFolderPath);
+		fs.mkdirSync(dataFolderPath);
+		fs.mkdirSync(contentFolderPath);
+		fs.writeFileSync(configFile, 'title: Test Book');
+		fs.writeFileSync(tableOfContentsFile, '- title: Notebook\n  url: /notebook\n- title: Markdown\n  url: /markdown\n- title: GitHub\n  url: https://github.com/\n  external: true');
+		fs.writeFileSync(notebookFile, '');
+		fs.writeFileSync(markdownFile, '');
+		mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
+		bookTreeViewProvider = new BookTreeViewProvider(rootFolderPath, mockExtensionContext.object);
+	});
+
+	it('should return all book nodes when element is undefined', async function (): Promise<void> {
+		const children = await bookTreeViewProvider.getChildren();
+		should(children).Array();
```

Comment:
nitpick: style for should assertions tends to be `should(children).be.Array()` so it reads as expected. It's syntactic sugar but makes reading it easier.